pkg/cpio/cpio.go

```diff
@@ -77,9 +78,28 @@ func (r Reader) ReadRecords() ([]Record, error) {
 
 type Writer struct {
 	rw RecordWriter
+	// There seems to be no harm done in stripping
+	// duplicate names when the record is written,
+	// and lots of harm done if we don't do it.
+	alreadyWritten map[string] bool
```

Comment:
nit: make this map[string]struct{}

empty struct takes no space, so that makes this map effectively a set. map[key]struct{} is the idiomatic way to do a set in go, and then you just use ```if _, ok := alreadyWritten[name]; ok {``` to test for presence of the key.